### ISSUE TITLE
How to get list of events via API

### ISSUE THREAD

**Issue by vkuznet** [indico/indico#1772]
Hello indico developers, I'm working on analytics project and would like to analyze how physicists events, i.e. conferences, influence physicists activities. To do so, I want to get list of past/future events for specific time range. I look-up existing API and don't see any way to do so. The REST API allow to get info from specific category, e.g. https://indico.cern.ch/export/categ/4648.json will return data from category categ for given location 4648. The question is how to discovery events within categories? Any RESTFull service should allow to get full list of elements within a group, but if I use https://indico.cern.ch/export/categ/ I get "Page not found", if I want to discover categories, i.e. https://indico.cern.ch/export/ I also get Page not found. It seems to me that API usage is quite useless at the moment within proper discovery of events. May be I am missing something and I'll be happy to get proper directions. Otherwise, my question to developers is the following. Is there any way to get list of events for specific time frame, sort of like calendar output on main CERN indico page? Thanks, Valentin

**Comment by ThiefMaster** [comment]
Check our api documentation:
http://indico.readthedocs.org/en/v1.9/http_api/

Unfortunately it's not really RESTful...

**Comment by vkuznet** [comment]
That's what I was reading, and that is why I posted the question. The question remains, can you implement REST part and allow to discover events?

**Comment by ThiefMaster** [comment]
To get all categories, use the top-level category (`/export/categ/0.json`). For restricting the results by date, see http://indico.readthedocs.org/en/v1.9/http_api/common/

**Comment by pferreir** [comment]
I agree that the API is incomplete when it comes to discovering categories. That's something to be fixed.
Still, you will be able to infer a lot about the category tree by analysing `additionalInfo.eventCategories` for `/export/categ/0.json`.
